File: DebriefRepository/src/main/java/org/example/debriefrepository/entity/Mission.java
package org.example.debriefrepository.entity;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;

import java.time.LocalDate;
import java.util.UUID;

@Getter
@Setter
@Entity
@Table(name = "mission", schema = "debrief_mgmt")
public class Mission {

    @Id
    @Column(name = "id", nullable = false, length = Integer.MAX_VALUE)
    private String id = UUID.randomUUID().toString();

    @Column(name = "content", nullable = false, length = Integer.MAX_VALUE)
    private String content;

    @Column(name = "start_date", nullable = false)
    private LocalDate startDate;

    @Column(name = "deadline", nullable = false)
    private LocalDate deadline;

    @ManyToOne(fetch = FetchType.LAZY, optional = false)
    @JoinColumn(name = "user_id", nullable = false)
    private User user;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "lesson_id")
    private Lesson lesson;

    @ManyToOne(fetch = FetchType.LAZY, optional = false)
    @JoinColumn(name = "debrief_id", nullable = false)
    private Debrief debrief;

}

File: DebriefRepository/src/main/java/org/example/debriefrepository/entities/Mission.kt
package org.example.debriefrepository.entities

import jakarta.persistence.*
import java.time.LocalDate

@Entity
@Table(name = "missions", schema = "debrief_mgmt")
open class Mission {
    @Id
    @Column(name = "id", nullable = false)
    open var id: Long? = null

    @Column(name = "content", nullable = false, length = Integer.MAX_VALUE)
    open var content: String? = null

    @Column(name = "start_date", nullable = false)
    open var startDate: LocalDate? = null

    @Column(name = "deadline", nullable = false)
    open var deadline: LocalDate? = null

    @ManyToMany(mappedBy = "missions")
    open var debriefs: MutableSet<Debrief> = mutableSetOf()

    @ManyToMany(mappedBy = "missions")
    open var lessons: MutableSet<Lesson> = mutableSetOf()

    @ManyToMany(mappedBy = "missions")
    open var users: MutableSet<org.example.debriefrepository.entities.User> = mutableSetOf()
}